Cleanups for mina-signer (#1414)
o1-labs/o1js: README-dev.md

npm run test:unit
```

> #### Running Mina-Signer Tests

Author: So there are a few ways you could go about writing this into the readme-dev. Essentially it's the fact that

`npm run build` 

is not sufficient, and makes rise 140 above incorrect. `npm run test` will fail once it gets through the `/lib` tests and arrives at the `/mina-signer` tests. as @mitschabaude pointed out, the `ci` moves into `/mina-signer` and runs build again before running `npm run test`. 

perhaps this change should be behind a header for running mina-signer tests and should simply be at the topmost reference to building. it's not as svelte however

 ```
npm run build
cd src/mina-signer
npm run build
cd ../..
```

so i stuck it here. 🤷‍♂️

Reviewer: That works well!

> #### Running Mina-Signer Tests
> `npm run build` is not recursive, thus in order for the Mina Signer tests to run you must execute the following from the root directory:

Reviewer: ```suggestion
In order for the mina-signer tests to run you must also build from inside its subdirectory:
```

Removing the header since the other kinds of tests below don't have one either

```sh
cd src/mina-signer
npm run build
cd ../..
```

This runs all the unit tests and provides you with a summary of the test results.

> Note that you can run individual jest tests via the command:

Author: optional comment here but I can't imagine outside devs don't ask this. could save some time. would have saved me some digging. up to you however.

Reviewer: Useful thanks!

```sh

> NODE_OPTIONS=--experimental-vm-modules npx jest <path/to/test.ts>

Reviewer: ```suggestion
./jest <path/to/test.ts>
```
we actually have a little script to make this not so hard to remember
